test/unit/batching/vendor-lines/get-payment-type.test.js

```diff
@@ -0,0 +1,108 @@
+const { BPS, CS, FDMR, LumpSums, SFI, SFIP, VetVisits, IMPS, ES } = require('../../../../app/constants/schemes')
+const { getPaymentType } = require('../../../../app/batching/vendor-lines/get-payment-type')
+
+describe('get payment type tests', () => {
+  let paymentType
+
+  test('return 1 when schemeId is CS and paymentType is 1', () => {
+    paymentType = '1'
+    const result = getPaymentType(CS, paymentType)
+    expect(result).toBe(1)
+  })
+
+  test('return 0 when schemeId is CS and paymentType is 0', () => {
+    paymentType = '0'
+    const result = getPaymentType(CS, paymentType)
+    expect(result).toBe(0)
+  })
+
+  test('return 0 when schemeId is CS and paymentType is 2', () => {
+    paymentType = '0'
+    const result = getPaymentType(CS, paymentType)
+    expect(result).toBe(0)
+  })
+
+  test('return "" when schemeId is BPS and paymentType is undefined', () => {
+    paymentType = undefined
+    const result = getPaymentType(BPS, paymentType)
+    expect(result).toBe('')
+  })
+
+  test('return "" when schemeId is BPS and paymentType is "0" ', () => {
+    paymentType = '0'
+    const result = getPaymentType(BPS, paymentType)
+    expect(result).toBe('')
+  })
+
+  test('return "" when schemeId is FDMR and paymentType is undefined', () => {
+    paymentType = undefined
+    const result = getPaymentType(FDMR, paymentType)
+    expect(result).toBe('')
+  })
+
+  test('return "" when schemeId is FDMR and paymentType is "0" ', () => {
+    paymentType = '0'
+    const result = getPaymentType(FDMR, paymentType)
+    expect(result).toBe('')
+  })
+
+  test('return 0 when schemeId is LumpSums and paymentType is "0" ', () => {
+    paymentType = '0'
+    const result = getPaymentType(LumpSums, paymentType)
+    expect(result).toBe(0)
+  })
+
+  test('return 0 when schemeId is SFI and paymentType is "0" ', () => {
+    paymentType = '0'
+    const result = getPaymentType(SFI, paymentType)
+    expect(result).toBe(0)
+  })
+
+  test('return 0 when schemeId is SFIP and paymentType is "0" ', () => {
+    paymentType = '0'
+    const result = getPaymentType(SFIP, paymentType)
+    expect(result).toBe(0)
+  })
+
+  test('return 0 when schemeId is VetVisits and paymentType is "0" ', () => {
+    paymentType = '0'
+    const result = getPaymentType(VetVisits, paymentType)
+    expect(result).toBe(0)
+  })
+
+  test('return "" when schemeId is ES and paymentType is "0"', () => {
+    paymentType = '0'
+    const result = getPaymentType(ES, paymentType)
+    expect(result).toBe('')
+  })
+
+  test('return "" when schemeId is ES and paymentType is "1"', () => {
+    paymentType = '1'
+    const result = getPaymentType(ES, paymentType)
+    expect(result).toBe('')
+  })
+
+  test('return "" when schemeId is ES and paymentType is undefined', () => {
+    paymentType = undefined
+    const result = getPaymentType(ES, paymentType)
+    expect(result).toBe('')
+  })
```

Comment:
I think this test should be before the other two tests for ES, as the match the same format as BPS and FDMR which also have tests for undefined

Reply:
I thought it would be better to keep them grouped by scheme rather than output, so it's easier to find an expected result for a particular payment.  Rather than the secondary input of payment type.  Paricularly as payment type only really changes for CS.

What do you think?

Reply:
Unless I'm misunderstanding what you're saying, I mean to keep the ES tests together, but have the undefined one appear first, the 0 and then 1. Same for IMPS

Reply:
No I misunderstood you, though you meant move all the undefineds for every scheme together, rather than reordering within scheme.

Reordered.